operators/pkg/controller/common/annotation/controller_version.go

```diff
@@ -0,0 +1,42 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package annotation
+
+import (
+	"github.com/elastic/cloud-on-k8s/operators/pkg/utils/k8s"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ControllerVersionAnnotation is the annotation name that indicates the last controller version to update a resource
+const ControllerVersionAnnotation = "k8s.elastic.co/controller-version"
+
+// UpdateControllerVersion updates the controller version annotation to the current version if necessary
+func UpdateControllerVersion(client k8s.Client, obj runtime.Object, version string) error {
+	metaObject, err := meta.Accessor(obj)
+	if err != nil {
+		log.Error(err, "error converting runtime object to metav1 object")
+		return err
+	}
+	annotations := metaObject.GetAnnotations()
+	if annotations == nil {
+		annotations = make(map[string]string)
+	}
+
+	// do not send extraneous update if the value would not change
+	if annotations[ControllerVersionAnnotation] == version {
```

Comment:
Should this not also return if there is any version set?

Reply:
I thought we would want to use this func because the caller decided that we do want to proceed with an update -- for instance it might be okay with going from 0.9.0 to 0.9.1. We might need additional funcs before calling this